test/RelayHub.test.ts

```diff
@@ -0,0 +1,1460 @@
+      });
+    });
+
+    context('with staked and registered relay', function () {
+      const url = 'http://relay.com';
+      const message = 'Enveloping RelayHub';
+      const messageWithNoParams = 'Method with no parameters';
+
+      let relayRequestEncodedFn: RelayRequest;
+      let signatureWithPermissiveVerifier: string;
+      let misbehavingVerifier: TestVerifierConfigurableMisbehavior;
+
+      beforeEach(async function () {
+        await relayHub
+          .connect(relayOwner)
+          .stakeForAddress(relayManager.address, 1000, {
+            value: ethers.utils.parseEther('2'),
+          });
+
+        const encodedFunction = recipient.interface.encodeFunctionData(
+          'emitMessage',
+          [message]
+        );
+
+        await relayHub
+          .connect(relayManager)
+          .addRelayWorkers([relayWorker.address]);
+        await relayHub.connect(relayManager).registerRelayServer(url);
+
+        relayRequestEncodedFn = cloneRelayRequest({
+          request: { data: encodedFunction },
+        });
+        ({ signature: signatureWithPermissiveVerifier } =
+          await getSuffixDataAndSignature(
+            forwarder,
+            relayRequestEncodedFn,
+            owner
+          ));
+
+        misbehavingVerifier = await deployContract(
+          'TestVerifierConfigurableMisbehavior'
+        );
+      });
+
+      context('with relay worker that is not an EOA', function () {
+        it('should not accept relay requests', async function () {
+          const signature = '0xdeadbeef';
+
+          const testRelayWorkerContract = await ethers
+            .getContractFactory('TestRelayWorkerContract')
+            .then((contract) => contract.deploy());
+          await relayHub
+            .connect(relayManager)
+            .addRelayWorkers([testRelayWorkerContract.address]);
+
+          const relayCall = testRelayWorkerContract.relayCall(
+            relayHub.address,
+            relayRequestEncodedFn,
+            signature,
+            {
+              gasPrice,
+            }
+          );
+
+          await expect(relayCall).to.be.rejectedWith(
+            'RelayWorker cannot be a contract'
+          );
+        });
+      });
+
+      context('with view functions only', function () {
+        it("should get 'verifierAccepted = true' and no revert reason as view call result of 'relayCall' for a valid transaction", async function () {
+          const relayCallView = await relayHub
+            .connect(relayWorker)
+            .callStatic.relayCall(
+              relayRequestEncodedFn,
+              signatureWithPermissiveVerifier,
+              {
+                gasPrice,
+              }
+            );
+
+          expect(relayCallView).to.be.true;
+        });
+
+        it.skip("should get Verifier's reject reason from view call result of 'relayCall' for a transaction with a wrong signature", async function () {
```

Comment:
Although I managed to get it work, I think we can remove it because it isn't necessary any more.